utils/httpsignature/hs2019.go

```diff
@@ -0,0 +1,27 @@
+package httpsignature
+
+import (
+	"crypto"
+	"crypto/hmac"
+	"crypto/sha512"
+	"crypto/subtle"
+)
+
+type HMACKey string
```

Comment:
I had chosen to use the new `SignRequest()` method to ensure compatibility with any existing code of yours. I am open to any changes that you propose but to keep things simple it might be smart to apply the new solutions to an entirely new codepath all together and leave `Sign()` as is